waiter/integration/waiter/request_timeout_test.clj

```diff
@@ -64,29 +64,37 @@
     (log/info "response: " (:body response))
     response))
 
-(deftest ^:parallel ^:integration-fast test-request-client-timeout
+(deftest ^:parallel ^:integration-fast test-backend-request-errors
   (testing-using-waiter-url
-    (log/info (str "request-client-timeout-test: if we can't get an instance quickly inside client timeout"))
-    (let [timeout-period 2000
-          extra-headers {:x-waiter-name (rand-name)
-                         :x-waiter-timeout timeout-period
-                         :x-waiter-debug "true"
-                         :x-kitchen-delay-ms (+ 2000 timeout-period)}
-          response (make-kitchen-request waiter-url extra-headers)
-          response-headers (:headers response)
-          response-body (:body response)
-          service-id (retrieve-service-id waiter-url (:request-headers response))]
-      (assert-response-status response 503)
-      (log/info "Response code check executed.")
-      (is (str/includes? response-body "Connection error while sending request to instance"))
-      (is (str/includes? response-body "onIdleExpired"))
-      (log/info "Response body check executed.")
-      (is (not (str/blank? (get response-headers "x-waiter-backend-id"))))
-      (is (not (str/blank? (get response-headers "x-waiter-backend-host"))))
-      (is (not (str/blank? (get response-headers "x-waiter-backend-port"))))
-      (is (not (str/blank? (get response-headers "x-waiter-backend-proto"))))
-      (is (not (str/blank? (get response-headers "x-cid"))))
-      (log/info "Response headers check executed.")
+    (let [assert-headers (fn [{:keys [headers]}]
+                           (is (not (str/blank? (get headers "x-waiter-backend-id"))))
+                           (is (not (str/blank? (get headers "x-waiter-backend-host"))))
+                           (is (not (str/blank? (get headers "x-waiter-backend-port"))))
+                           (is (not (str/blank? (get headers "x-waiter-backend-proto"))))
+                           (is (not (str/blank? (get headers "x-cid")))))
```

Comment:
I would favor either inlining these or making this a macro for better IDE support